packages/backend/src/logging/caslAuditWrapper.ts

```diff
@@ -0,0 +1,201 @@
+import { type Ability } from '@casl/ability';
+import type { Rule as CaslRule } from '@casl/ability/dist/types/Rule';
+import { Abilities, type AnyObject } from '@casl/ability/dist/types/types';
+import { type SessionUser } from '@lightdash/common';
+import {
+    AuditActor,
+    AuditContext,
+    AuditResource,
+    AuditStatusType,
+    createAuditLogEvent,
+    type AuditLogEvent,
+} from './auditLog';
+
+export type AuditLogger = (event: AuditLogEvent) => void;
+
+export type AuditableUser = Pick<
+    SessionUser,
+    | 'userUuid'
+    | 'email'
+    | 'firstName'
+    | 'lastName'
+    | 'organizationUuid'
+    | 'role'
+>;
+
+type AuditHelperArgs = {
+    user: AuditableUser;
+    action: string;
+    subject: AnyObject;
+    ip?: string;
+    userAgent?: string;
+    requestId?: string;
+    ruleConditions?: string;
+};
+
+/**
+ * Creates an audit actor from a user
+ */
+const createActorFromUser = (user: AuditableUser): AuditActor => ({
+    id: user.userUuid,
+    email: user.email || '',
+    firstName: user.firstName || '',
+    lastName: user.lastName || '',
+    organizationUuid: user.organizationUuid || '',
+    organizationRole: user.role || 'unknown',
+    // TODO: Add group memberships
+    groupMemberships: [],
+});
+
+const createResourceFromSubject = (subject: AnyObject): AuditResource => {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
Should we type the subject arg as `ForcedSubject<String>` to guarantee it has this property?

Reply:
Much better thanks!